packages/@react-stately/toast/src/useToastState.ts

```diff
@@ -159,30 +155,27 @@ export class ToastQueue<T> {
     if (index >= 0) {
       this.queue[index].onClose?.();
       this.queue.splice(index, 1);
+      let closedToast = this.visibleToasts.find(t => t.key === key);
+      if (closedToast) {
+        closedToast.animation = 'exiting';
```

Comment:
wonder if there will be problems due to mutating the toast, which is passed into props of a component...